File: src/fixtures/bun/strutil.ts
export function reverse(s: string): string {
	return s.split("").reverse().join("");
}

export function capitalize(s: string): string {
	if (s.length === 0) {
		return s;
	}
	return s[0].toUpperCase() + s.slice(1);
}

export function truncate(s: string, maxLen: number): string {
	if (s.length <= maxLen) {
		return s;
	}
	return s.slice(0, maxLen) + "...";
}

export function padLeft(s: string, len: number, ch: string): string {
	if (s.length >= len) {
		return s;
	}
	return ch.repeat(len - s.length) + s;
}

File: src/fixtures/goproject/strutil.go
package fixture

import "strings"

func Reverse(s string) string {
	runes := []rune(s)
	for i, j := 0, len(runes)-1; i < j; i, j = i+1, j-1 {
		runes[i], runes[j] = runes[j], runes[i]
	}
	return string(runes)
}

func Capitalize(s string) string {
	if len(s) == 0 {
		return s
	}
	return strings.ToUpper(s[:1]) + s[1:]
}

func Truncate(s string, maxLen int) string {
	if len(s) <= maxLen {
		return s
	}
	return s[:maxLen] + "..."
}

func PadLeft(s string, width int, pad rune) string {
	for len(s) < width {
		s = string(pad) + s
	}
	return s
}
